### ISSUE TITLE
[FontBakeryCondition:api_gfonts_ttFont]: urlopen error [SSL: CERTIFICATE_VERIFY_FAILED]

### ISSUE THREAD

**Issue by mjlagattuta** [googlefonts/fontbakery#2274]
original title of this issue was:
Checks 117–119, 154: Unable to get info from hosted fonts
----

## Observed behaviour

When running fontbakery on Cabin and Expletus Sans, I end up with 4 errors all with an SSL certificate issue and the same traceback:

```
>> com.google.fonts/check/154 with (('font[0]', 'Cabin-Regular.ttf'),)
   Check font has same encoded glyphs as version hosted on fonts.google.com
 * ERROR: The condition <FontBakeryCondition:api_gfonts_ttFont> had an error: FailedConditionError: The condition <FontBakeryCondition:remote_styles> had an error: URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificateverify failed (_ssl.c:841)>
           File "/Users/lags/Desktop/fontbakery-venv/lib/python3.6/site-packages/fontbakery/checkrunner.py", line 370, in _evaluate_condition
             args = self._get_args(condition, iterargs, path)
           File "/Users/lags/Desktop/fontbakery-venv/lib/python3.6/site-packages/fontbakery/checkrunner.py", line 509, in _get_args
             args[name] = self._get(name, iterargs, path)
           File "/Users/lags/Desktop/fontbakery-venv/lib/python3.6/site-packages/fontbakery/checkrunner.py", line 483, in _get
             raise error


   Result: ERROR
```

This happens when running fontbakery on a font file when there is also a METADATA.pb file in the same directory, but if I remove the METADATA.pb file, then I only get an error for check 117.

It is very possible I may be doing something wrong on my end, though I am not sure what that would be; my setup is very straightforward: I am running fontbakery v0.6.5 from a virtual environment.

## Expected behaviour

I would expect METADATA.pb to not affect checks that don't relate to it

## Resources and exact process needed to replicate

The exact location I was running fontbakery on can be found [here](https://github.com/mjlagattuta/fonts/tree/cabin/ofl/cabin)

**Comment by felipesanches** [comment]
What caught my attention was `URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificateverify failed (_ssl.c:841)>`...

This does not happen here on my computer.

@mjlagattuta, Are you still having this problem?
Please, re-open the issue if the problem still persists.

**Comment by mjlagattuta** [comment]
Reopening as I am still having this issue.

> What caught my attention was URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificateverify failed (_ssl.c:841)>...

I am not sure why the url would be problematic. I have tested my connection via `ping fonts.google.com` and everything appears to be fine.

**Comment by felipesanches** [comment]
Could you please install ipython (with `pip install ipython`) and then inserting the following sequence of commands?

<img width="491" alt="captura de tela 2019-01-29 as 17 51 51" src="https://user-images.githubusercontent.com/213676/51936180-bd804700-23ee-11e9-97d1-88cfb7968e3f.png">

Let me know if you get the same results, or if it crashes at some point.

**Comment by mjlagattuta** [comment]
Error at [3]
![screen shot 2019-01-29 at 3 05 24 pm](https://user-images.githubusercontent.com/35570153/51936952-5905bd80-23d7-11e9-85d9-152889bd5c1a.png)

**Comment by felipesanches** [comment]
It puzzles me why would the SSL certificate fail to be verified in your computer, but works nicely on mine.

Can you try to create a new virtual environment and re-install fontbakery and its dependencies and then verify whether you still get the same error?

you can do it this way:

```
virtualenv newvenv
. newvenv/bin/activate
pip install fontbakery
pip install ipython
```

and then re-run that ipython session to see if the problem persists.

**Comment by mjlagattuta** [comment]
Setup a new venv
```
python3 -m venv newvenv
. newvenv/bin/activate
pip install fontbakery
pip install ipython
```
and got the same error

**Comment by mjlagattuta** [comment]
That was it, works perfectly now.

Thanks @anthrotype!

**Comment by felipesanches** [comment]
Let's try to catch the exception and emit an ERROR message explaining what the user needs to do.

Can you help me craft a concise, but still helpful & user-friendly message?

**Comment by felipesanches** [comment]
@mjlagattuta I have added a message explaining the issue to the users. Is there any way we could replicate the problem in order to test it?

**Comment by felipesanches** [comment]
well.. for now I will close this. Feel free to reopen if any further related issue resurface.

**Comment by anthrotype** [comment]
For reference, this is a screenshot of the "python-3.7.2-macosx10.9.pkg" installer from Python.org, and the relevant paragraph from the installer's info page:

![image](https://user-images.githubusercontent.com/6939968/52296959-eae56b80-297f-11e9-8ce2-335af84c29b6.png)

> Certificate verification and OpenSSL
> This variant of Python 3.7 includes its own private copy of OpenSSL 1.1.0.  The deprecated Apple-supplied OpenSSL libraries are no longer used.  This means that the trust certificates in system and user keychains managed by the Keychain Access application and the security command line utility are no longer used as defaults by the Python ssl module.  A sample command script is included in /Applications/Python 3.7 to install a curated bundle of default root certificates from the third-party certifi package (https://pypi.org/project/certifi/).  If you choose to use certifi, you should consider subscribing to the project's email update service to be notified when the certificate bundle is updated.
> The bundled pip included with this installer has its own default certificate store for verifying download connections.

Even when the installation is completed, the installer reminds the user to double-click on the bundled "Install Certificates.command" shell, found in "/Applications/Python 3.7" folder.
